src/libstore/derivations.cc

```diff
@@ -272,7 +272,15 @@ Derivation parseDerivation(const Store & store, std::string && s, std::string_vi
 
 static void printString(string & res, std::string_view s)
 {
-    char buf[s.size() * 2 + 2];
+    char * buf;
+    size_t bufsize = s.size() * 2 + 2;
+    std::string dynbuf;
```

Comment:
please use a `std::unique<char[]>` instead. zero-initializing (as string::resize does) is an unnecessary cost here